code/modules/mob/living/basic/lavaland/node_drone/node_drone.dm

```diff
@@ -0,0 +1,149 @@
+ * If the drone survives, they will fly away to safety as the vent spawns ores.
+ * If the drone dies, the wave defense will fail.
+ */
+
+/mob/living/basic/node_drone
+	name = "NODE drone"
+	desc = "Standard in-atmosphere drone, used by Nanotrasen to operate and excavate valuable ore vents."
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "mining_node_active"
+	icon_living = "mining_node_active"
+	icon_dead = "mining_node_active"
+
+	maxHealth = 500
+	health = 500
+	density = TRUE
+	pass_flags = PASSTABLE|PASSGRILLE|PASSMOB
+	mob_size = MOB_SIZE_LARGE
+	mob_biotypes = MOB_ROBOTIC
+	faction = list(FACTION_STATION, FACTION_NEUTRAL)
+	light_range = 4
+
+	speak_emote = list("chirps")
+	response_help_continuous = "pets"
+	response_help_simple = "pet"
+	response_disarm_continuous = "gently pushes aside"
+	response_disarm_simple = "gently push aside"
+	response_harm_continuous = "clangs"
+	response_harm_simple = "clang against"
+
+	ai_controller = /datum/ai_controller/basic_controller/node_drone
+
+	/// Is the drone currently attached to a vent?
+	var/active_node = FALSE
+	/// What status do we currently track for icon purposes?
+	var/flying_state = NEUTRAL_STATE
+	/// Weakref to the vent the drone is currently attached to.
+	var/obj/structure/ore_vent/attached_vent = null
+	/// Set when the drone is begining to leave lavaland after the vent is secured.
+	var/escaping = FALSE
+
+/mob/living/basic/node_drone/death(gibbed)
+	. = ..()
+	explosion(origin = src, light_impact_range = 1, smoke = 1)
+	qdel(src)
+
+/mob/living/basic/node_drone/Destroy()
+	attached_vent?.node = null //clean our reference to the vent both ways.
+	attached_vent = null
+	return ..()
+
+
+/mob/living/basic/node_drone/examine(mob/user)
+	. = ..()
+	var/sameside = user.faction_check_atom(src, exact_match = FALSE)
+	if(sameside)
+		. += span_notice("This drone is currently attached to a mineral vent. You should protect it from harm to secure the mineral vent.")
+	else
+		. += span_warning("This vile Nanotrasen trash is trying to destroy the environment. Attack it to free the mineral vent from its grasp.")
+
+/mob/living/basic/node_drone/update_icon_state()
+	. = ..()
+
+	icon_state = "mining_node_active"
+
+	if(flying_state == FLY_IN_STATE || flying_state == FLY_OUT_STATE)
+		icon_state = "mining_node_flying"
+
+/mob/living/basic/node_drone/proc/arrive(obj/structure/ore_vent/parent_vent)
+	attached_vent = parent_vent
+	flying_state = FLY_IN_STATE
+	update_appearance(UPDATE_ICON_STATE)
+	pixel_z = 400
+	animate(src, pixel_z = 0, time = 2 SECONDS, easing = QUAD_EASING|EASE_OUT, flags = ANIMATION_PARALLEL)
+
+
+/**
+ * Called when wave defense is completed. Visually flicks the escape sprite and then deletes the mob.
+ */
+/mob/living/basic/node_drone/proc/escape()
+	var/funny_ending = FALSE
+	attached_vent?.unbuckle_mob(src)
+	if(!escaping)
+		escaping = TRUE
+		flick("mining_node_escape", src)
+		addtimer(CALLBACK(src, PROC_REF(escape)), 1.9 SECONDS)
+		return
```

Comment:
Why is this formatted this way instead of being two procs?

Reply:
I tried being clever here, but you're right that this isn't very elegant in hindsight.